intel_extension_for_transformers/llm/runtime/graph/models/baichuan/baichuan_utils.cpp

```diff
@@ -0,0 +1,192 @@
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <algorithm>
+#include <cassert>
+#include <cinttypes>
+#include <cstring>
+#include <exception>
+#include <fstream>
+#include <iterator>
+#include <memory>
+#include <random>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "core/data_types.h"
+#include "core/ne.h"
+#include "core/ne_layers.h"
+#include "models/baichuan/baichuan.h"
+#include "models/model_utils/model_config.h"
+#include "models/model_utils/model_files.h"
+#include "models/model_utils/model_types.h"
+#include "models/model_utils/model_utils.h"
+#include "models/model_utils/util.h"
+#include "models/models.h"
+
+void model_load_internal(const std::string& fname, model_archs arch, model_context& lctx, int n_ctx, int n_gpu_layers,
+                         bool use_mmap, bool use_mlock, bool vocab_only, model_progress_callback progress_callback,
+                         void* progress_callback_user_data) {
+  lctx.t_start_us = ne_time_us();
+
+  std::unique_ptr<IModel> ms(new BAICHUAN());
+  ms->init(fname.c_str(), lctx, n_ctx, n_gpu_layers, use_mmap, use_mlock, vocab_only);
+  ms->load(lctx, progress_callback, progress_callback_user_data);
+
+  lctx.t_load_us = ne_time_us() - lctx.t_start_us;
+}
+
+void BAICHUAN::init(const char* path_model, model_context& lctx, int n_ctx_, int n_gpu_layer_, bool use_mmap_,
+                    bool use_mlock_, bool vocab_only_) {
+  n_ctx = n_ctx_;
+  n_gpu_layer = n_gpu_layer_;
+  use_mmap = use_mmap_;
+  use_mlock = use_mlock_;
+  vocab_only = vocab_only_;
+  auto& model = lctx.model;
+  ml.reset(new model_model_loader(path_model, false, vocab_only));
+  lctx.vocab = std::move(ml->file_loaders.at(0)->vocab);
+  model.hparams = ml->file_loaders.at(0)->hparams;
+  model_file_version file_version = ml->file_loaders.at(0)->file_version;
+  auto& hparams = model.hparams;
+  n_ff = 4 * hparams.n_embd;
+  hparams.n_ctx = n_ctx;
+  fprintf(stderr, "%s: n_vocab    = %u\n", __func__, hparams.n_vocab);
+  fprintf(stderr, "%s: n_ctx      = %u\n", __func__, hparams.n_ctx);
+  fprintf(stderr, "%s: n_embd     = %u\n", __func__, hparams.n_embd);
+  fprintf(stderr, "%s: n_mult     = %u\n", __func__, hparams.n_mult);
+  fprintf(stderr, "%s: n_head     = %u\n", __func__, hparams.n_head);
+  fprintf(stderr, "%s: n_layer    = %u\n", __func__, hparams.n_layer);
+  fprintf(stderr, "%s: n_rot      = %u\n", __func__, hparams.n_rot);
+  fprintf(stderr, "%s: n_ff       = %u\n", __func__, n_ff);
+  fprintf(stderr, "%s: n_parts    = %zu\n", __func__, ml->file_loaders.size());
+  n_embd = hparams.n_embd;
+  n_vocab = hparams.n_vocab;
+  n_layer = hparams.n_layer;
+  scratch = baichuan_mem_req(n_layer);
+  model.scratchs = scratch;
+}
+
+#define MODEL_BACKEND_OFFLOAD NE_BACKEND_CPU
```

Comment:
useless?

Reply:
All xxx_utils.cpp files have this macro declaration.